src/middle-tier/ssoauth-helper.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See full license in root of repo. -->
+ *
+ * This file defines the routes within the authRoute router.
+ */
+
+import fetch from "node-fetch";
+import form from "form-urlencoded";
+import jwt from "jsonwebtoken";
+import { JwksClient } from "jwks-rsa";
+
+/* global process, console */
+
+const DISCOVERY_KEYS_ENDPOINT = "https://login.microsoftonline.com/common/discovery/v2.0/keys";
+
+export async function getAccessToken(authorization) {
+  if (!authorization) {
+    let error = new Error("No Authorization header was found.");
+    return Promise.reject(error);
+  } else {
+    const scopeName = process.env.SCOPE || "User.Read";
+    const [, /* schema */ assertion] = authorization.split(" ");
+
+    const tokenScopes = jwt.decode(assertion).scp.split(" ");
+    const accessAsUserScope = tokenScopes.find((scope) => scope === "access_as_user");
+    if (!accessAsUserScope) {
+      throw new Error("Missing access_as_user");
+    }
+
+    const formParams = {
+      client_id: process.env.CLIENT_ID,
+      client_secret: process.env.CLIENT_SECRET,
+      grant_type: "urn:ietf:params:oauth:grant-type:jwt-bearer",
+      assertion: assertion,
+      requested_token_use: "on_behalf_of",
+      scope: [scopeName].join(" "),
+    };
+
+    const stsDomain = "https://login.microsoftonline.com";
+    const tenant = "common";
+    const tokenURLSegment = "oauth2/v2.0/token";
+    const encodedForm = form(formParams);
+
+    const tokenResponse = await fetch(`${stsDomain}/${tenant}/${tokenURLSegment}`, {
```

Comment:
Would be nice to use the MSAL library because it will cache the OBO Graph token for you. For anyone going to deployment, this could be a performance hit and they may be tempted to roll their own caching, which would be a bad idea. It's difficult to do properly.

Reply:
Not something I'm going to mess with right now.  This is what was being used before and other than potentially caching that you point out I'm not sure what other differences there are between the two approaches.  This change could be done later after we get these basic changes out.